### PR TITLE
SRX-GUPKQB - argocd baseurl argument

### PR DESCRIPTION
**Pull request by medo-freiheit**
* clean up unused Argocd arguments
* add Argocd baseurl argument
* create getConfig rpc

### CODE REVIEW

**Comment by mouhsen-ibrahim** [comment]
@medo-freiheit please remove installing the ArgoCD client in Dockerfile

File: pkg/api/api.proto [hunk 1]
```diff
@@ -109,10 +109,15 @@ message CreateEnvironmentRequest {
 }
 
 service OverviewService {
+  rpc GetConfig (google.protobuf.Empty) returns (GetConfigResponse) {}
   rpc GetOverview (GetOverviewRequest) returns (GetOverviewResponse) {}
   rpc StreamOverview (GetOverviewRequest) returns (stream GetOverviewResponse) {}
 }
 
+message GetConfigResponse {
+  string argocdBaseUrl = 1;
```
Comment (hannesg) [inline pkg/api/api.proto:118]: I would suggest that we make an argoCd sub-message. Then we can group all argoCd related configs.

File: charts/kuberpult/templates/cd-service.yaml [hunk 2]
```diff
@@ -80,6 +80,8 @@ spec:
           value: {{ required ".Values.git.url is required" .Values.git.url | quote }}
         - name: KUBERPULT_GIT_BRANCH
           value: {{ .Values.git.branch | quote }}
+        - name: KUBERPULT_ARGOCD_BASE_URL
```
Comment (hannesg) [inline charts/kuberpult/templates/cd-service.yaml:83]: I would let the frontend-service handle this request as it is strictly for the frontend.

**Comment by medo-freiheit** [comment]
> @medo-freiheit please remove installing the ArgoCD client in Dockerfile



@mouhsen-ibrahim @hannesg  I think we still need it for syncing, right? [here](https://github.com/freiheit-com/kuberpult/blob/main/services/cd-service/pkg/service/service.go#L241)

**Comment by sven-urbanski-freiheit-com** [comment]
> > @medo-freiheit please remove installing the ArgoCD client in Dockerfile
> 
> @mouhsen-ibrahim @hannesg I think we still need it for syncing, right? [here](https://github.com/freiheit-com/kuberpult/blob/main/services/cd-service/pkg/service/service.go#L241)

We don't use the sync endpoint in any team, please remove it. @medo-freiheit

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**